File: app/src/main/java/com/commonpepper/photosen/ui/activities/PreviewActivity.java
package com.commonpepper.photosen.ui.activities;

import android.os.Bundle;
import android.view.View;
import android.widget.ProgressBar;
import android.widget.Toast;

import com.commonpepper.photosen.R;
import com.github.chrisbanes.photoview.PhotoView;
import com.squareup.picasso.Callback;
import com.squareup.picasso.Picasso;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

public class PreviewActivity extends AppCompatActivity {
    public static final String TAG_URL = "tag_url";

    private static final int MAX_WIDTH_PIXELS = 5000;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        getWindow().getDecorView().setSystemUiVisibility(
                View.SYSTEM_UI_FLAG_LAYOUT_STABLE
                        | View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
                        | View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN
                        | View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
                        | View.SYSTEM_UI_FLAG_FULLSCREEN
                        | View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY);
        setContentView(R.layout.activity_preview);

        PhotoView photoView = findViewById(R.id.preview_image);
        ProgressBar progressBar = findViewById(R.id.preview_progress);

        String url = getIntent().getStringExtra(TAG_URL);
        if (url != null) {
            Picasso.get().load(url).resize(MAX_WIDTH_PIXELS, 0).onlyScaleDown().into(photoView, new Callback() {
                @Override
                public void onSuccess() {
                    progressBar.setVisibility(View.GONE);
                }

                @Override
                public void onError(Exception e) {
                    progressBar.setVisibility(View.GONE);
                    Toast.makeText(PreviewActivity.this, getString(R.string.something_went_wrong), Toast.LENGTH_SHORT).show();
                }
            });
        } else {
            finish();
            Toast.makeText(this, getString(R.string.something_went_wrong), Toast.LENGTH_SHORT).show();
        }
    }
}

File: app/src/main/java/com/commonpepper/photosen/ui/activities/PreviewActivity.kt
package com.commonpepper.photosen.ui.activities

import android.os.Bundle
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.commonpepper.photosen.R.layout
import com.commonpepper.photosen.R.string
import com.squareup.picasso.Callback
import com.squareup.picasso.Picasso
import kotlinx.android.synthetic.main.activity_preview.*

class PreviewActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        window.decorView.systemUiVisibility = (View.SYSTEM_UI_FLAG_LAYOUT_STABLE
                or View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
                or View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN
                or View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
                or View.SYSTEM_UI_FLAG_FULLSCREEN
                or View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY)
        setContentView(layout.activity_preview)
        val url: String? = intent.getStringExtra(TAG_URL)
        if (url != null) {
            Picasso.get().load(url).resize(MAX_WIDTH_PIXELS, 0).onlyScaleDown().into(imagePreview, object : Callback {
                override fun onSuccess() {
                    progressPreview.visibility = View.GONE
                }

                override fun onError(e: Exception?) {
                    progressPreview.visibility = View.GONE
                    Toast.makeText(this@PreviewActivity, getString(string.something_went_wrong), Toast.LENGTH_SHORT).show()
                }
            })
        } else {
            finish()
            Toast.makeText(this, getString(string.something_went_wrong), Toast.LENGTH_SHORT).show()
        }
    }

    companion object {
        const val TAG_URL = "tag_url"
        private const val MAX_WIDTH_PIXELS = 5000
    }
}
